pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -423,25 +504,32 @@ private static void throwTableAccessError(TableAuthorizationResult tableAuthoriz
   }
 
   /**
-   * Runs the query planning in a separate thread so that we can enforce a timeout on it (in some rare cases,
-   * we can see query compilation taking a very long time).
+   * Calls the given callable in a separate thread and enforces a timeout on it.
+   *
+   * The only exception that can be thrown by this method is a QueryException. All other exceptions are caught and
+   * wrapped in a QueryException. Specifically, {@link TimeoutException} is caught and wrapped in a QueryException with
+   * the error code {@link QueryErrorCode#BROKER_TIMEOUT} and other exceptions are treated as internal errors.K
```

Comment:
```suggestion
   * the error code {@link QueryErrorCode#BROKER_TIMEOUT} and other exceptions are treated as internal errors.
```

Reply:
Strange. Mine is doing the same, but the exact text placed in another section is fine. It is probably a bug in how IntelliJ renders the new javadoc format. The [JEP-467](https://openjdk.org/jeps/467) defines a new way to define throws, parameters, etc sections, but it doesn't seem to be included in Java 23